pkg/provider/containergroup_to_pod.go

```diff
@@ -35,41 +37,45 @@ func (p *ACIProvider) containerGroupToPod(cg *azaci.ContainerGroup) (*v1.Pod, er
 	return updatedPod, nil
 }
 
-func (p *ACIProvider) getPodStatusFromContainerGroup(cg *azaci.ContainerGroup) (*v1.PodStatus, error) {
+func (p *ACIProvider) getPodStatusFromContainerGroup(ctx context.Context, cg *azaciv2.ContainerGroup) (*v1.PodStatus, error) {
+	logger := log.G(ctx).WithField("method", "getPodStatusFromContainerGroup")
+
 	// cg is validated
 	allReady := true
-	var firstContainerStartTime, lastUpdateTime metav1.Time
+	var firstContainerStartTime, lastUpdateTime time.Time
 
-	containerStatuses := make([]v1.ContainerStatus, 0, len(*cg.Containers))
-	containersList := *cg.Containers
+	containerStatuses := make([]v1.ContainerStatus, 0, len(cg.Properties.Containers))
+	containersList := cg.Properties.Containers
 
 	for i := range containersList {
-		err := validation.ValidateContainer(containersList[i])
+		err := validation.ValidateContainer(ctx, containersList[i])
+		logger.Debugf("container %s has missing fields. retrying the validation...", *containersList[i].Name)
 		if err != nil {
 			return nil, err
 		}
-		firstContainerStartTime := metav1.NewTime(containersList[0].InstanceView.CurrentState.StartTime.Time)
-		lastUpdateTime := firstContainerStartTime
+
+		firstContainerStartTime := containersList[0].Properties.InstanceView.CurrentState.StartTime
```

Comment:
This code looks wrong for lastUpdateTime, can you check the use of `lastUpdateTime` ?

Reply:
This is a default value only (old behavior) for the `lastUpdateTime`.  It then will get updated either by [getPodConditionsFromACIState](https://github.com/virtual-kubelet/azure-aci/pull/422/files#diff-5b6db607dc03cc9c58c4c465d78a1c87de2db52180ec5511c3f21d27c76ca647R188) or according to [containerStartTime](https://github.com/virtual-kubelet/azure-aci/pull/422/files#diff-5b6db607dc03cc9c58c4c465d78a1c87de2db52180ec5511c3f21d27c76ca647R77).

Reply:
Fixed.